components/Wallet/Wallet.tsx

```diff
@@ -19,11 +20,27 @@ import { WalletIcon } from "./WalletIcon";
 
 export function Wallet() {
   const [{ wallet, connecting }, connect] = useConnectWallet();
+  const [{ connectedChain }, setChain] = useSetChain();
   const connectedWallets = useWallets();
   const { setProvider, setSigner, setSigningKeys } = useWalletContext();
   const { setVoting, setVotingToken } = useContractsContext();
   const { openPanel } = usePanelContext();
   const { address, truncatedAddress } = useUserContext();
+  const { addErrorMessage, removeErrorMessage } = useErrorContext();
+
+  useEffect(() => {
+    if (!connectedChain) return;
```

Comment:
I added this to check that the logic for the generic data fetching error message works, and because people keep complaining about the errors when they connect to the wrong network lol